image-build/virt-customize/build.sh

```diff
@@ -1,18 +1,20 @@
 #!/usr/bin/env bash
 set -eEuo pipefail
 
-VERSION=k8s-v$CONFIG__KUBERNETES_VERSION-ubuntu-$CONFIG__UBUNTU_VERSION-amd64
+VERSION=k8s-v$KUBERNETES_VERSION-ubuntu-$UBUNTU_VERSION-$UBUNTU_RELEASE-amd64-$METALKAST_VERSION
 OUTPUT_DIR=output/$VERSION
-mkdir -p $OUTPUT_DIR
 
+printenv > printenv.txt
 BUILD_ENVIRONMENT_VERSION_FILE=md5sum.txt
+find -type f -not \( -path "./output/*" -o -name $BUILD_ENVIRONMENT_VERSION_FILE \) -print0  | xargs -0 md5sum | tee $BUILD_ENVIRONMENT_VERSION_FILE
```

Comment:
I don't think you can rely on the order `find` returns files - better to have `sort` (probably after `xargs ...`)
Also consider `shasum -a 256` instead of `md5sum`